lib/galaxy/webapps/galaxy/api/folder_contents.py

```diff
@@ -2,93 +2,99 @@
-        :param  payload:    dictionary structure containing:
-            :param from_hda_id:         (optional) the id of an accessible HDA to copy into the library
-            :type  from_hda_id:         encoded id
-            :param from_hdca_id:         (optional) the id of an accessible HDCA to copy into the library
-            :type  from_hdca_id:         encoded id
-            :param ldda_message:        (optional) the new message attribute of the LDDA created
-            :type   ldda_message:       str
-            :param extended_metadata:   (optional) dub-dictionary containing any extended metadata to associate with the item
-            :type  extended_metadata:   dict
-        :type   payload:    dict
 
-        :returns:   a dictionary describing the new item if ``from_hda_id`` is supplied or a list of
-                    such dictionaries describing the new items if ``from_hdca_id`` is supplied.
-        :rtype:     object
+@router.cbv
+class FastAPILibraryFoldersContents:
+    service: LibraryFolderContentsService = depends(LibraryFolderContentsService)
 
-        :raises:    ObjectAttributeInvalidException,
-            InsufficientPermissionsException, ItemAccessibilityException,
-            InternalServerError
+    @router.get(
+        "/api/folders/{folder_id}/contents",
+        summary="Returns a list of a folder's contents (files and sub-folders) with additional metadata about the folder.",
+        responses={
+            200: {
+                "description": "The contents of the folder that match the query parameters.",
+                "model": LibraryFolderContentsIndexResult,
+            },
+        },
+    )
+    def index(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        folder_id: EncodedDatabaseIdField = FolderIdPathParam,
+        limit: int = LimitQueryParam,
+        offset: int = OffsetQueryParam,
+        search_text: Optional[str] = SearchQueryParam,
+        include_deleted: Optional[bool] = IncludeDeletedQueryParam,
+    ):
+        """Returns a list of a folder's contents (files and sub-folders).
+
+        Additional metadata for the folder is provided in the response as a separate object containing data
+        for breadcrumb path building, permissions and other folder's details.
         """
-        return self.service.create(trans, encoded_folder_id, payload)
+        payload = LibraryFolderContentsIndexQueryPayload(
+            limit=limit, offset=offset, search_text=search_text, include_deleted=include_deleted
+        )
+        return self.service.index(trans, folder_id, payload)
+
+    @router.post(
+        "/api/folders/{folder_id}/contents",
+        name="add_history_datasets_to_library",
+        summary="Creates a new library file from an existing HDA/HDCA.",
+    )
+    def create(
+        self,
+        trans: ProvidesUserContext = DependsOnTrans,
+        folder_id: EncodedDatabaseIdField = FolderIdPathParam,
```

Comment:
```suggestion
        folder_id: DecodedDatabaseIdField = FolderIdPathParam,
```
that should give you integer primary keys right away

Reply:
Cool! I've pushed one commit with all the changes